test/compressed.js

```diff
@@ -0,0 +1,49 @@
+var test = require('tape');
+var zlib = require('zlib');
+var Graylog = require('..');
+
+test('compressed', function (t) {
+	var client = new Graylog({ servers: [{ host: '127.0.0.1', port: 12345 }], deflate: 'optimal' });
+
+	client.on('message', function (message) {
+		message = JSON.parse(zlib.inflateSync(message));
+
+		t.equal(message.short_message, 'short message');
+		t.equal(message.full_message, 'full message'.repeat(1000));
```

Comment:
True, but that pattern specifically is one you reuse frequently (see test/chunked.js, that line is everywhere).

Right now, you have the following data constructs repeated across numerous test files:

* short short message ('short mesage')
* short full message ('full message')
* long full message ('full message'.repeat(1000))

Which seems to indicate you wish to test against those specific data constructs. But if you wanted at a later time, say, to change that to random-generated strings, or to add different data constructs (to test string messages that break the current implementation so to ensure future implementations won't suffer from regression), then you'd end up doing a lot of manual work.

One thing which can be done to avoid that is to add those often used data constructs in their own file instead. This has the advantage of:

1. Documenting what data constructs should be used to test future features. Even if you'd have 100% code coverage, you could still request that all new tests for new features be executed against those data constructs (at the very least).

2. Easier to alter if the currently used value is deemed inadequate at some point